tutorials/kitties-part-01/3-step3.mdx

```diff
@@ -0,0 +1,1332 @@
+perform less read/writes to update multiple values. Read more about storage best practices 
+[here](https://doc.deepernetwork.org/v3/runtime/storage/#best-practices).
+
+#### What information to include?
+
+Let's first go over what information a single Kitty will carry:
+
+* `dna`: the hash used to identify the DNA of a Kitty, which corresponds to its unique features. 
+DNA is also used to breed new Kitties. An will be used as key in the storage.
+* `price`: this is a balance that corresponds to the amount needed to buy a Kitty and set by its owner.
+* `gender`: an enum that can be either Male or Female.
+* `owner`: an account ID designating a single owner.
+* `generation`: to keep track of the number of generations a Kitty has been bred for.
+
+#### Sketching out the types held by our struct
+
+Looking at the items of our struct from above, we can deduce the following types:
+
+* `[u8; 16]` for dna - to use 16 bytes to represent a Kitty's DNA.
+* `BalanceOf` for price - this is a custom type using FRAME's Currency trait.
+* `Gender` for gender - we are going to create this!
+* `u64` for generation.
+
+So we get the following as our struct kitty struct.
+
+Copy it to replace the TODO for the kitty struct:
+
+```rust
+// Struct for holding kitty information
+	#[derive(Clone, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+	#[scale_info(skip_type_params(T))]
+	pub struct Kitty<T: Config> {
+		// An array of 16 bytes representing the kitty's DNA.
+		pub dna: [u8; 16],
+
+		// The price of the kitty, stored as an optional balance.
+    	// If set to `None`, it assumes the kitty is not for sale.
+		pub price: Option<BalanceOf<T>>,
+
+		// The gender of the kitty.
+		pub gender: Gender,
+
+		// The owner of the kitty, stored as the account ID.
+		pub owner: T::AccountId,
+
+		// Generation of the kitty, stored as a u64.
+		pub generation: u64,
+	}
+```
+
+Notice how we use the derive macro to include various helper traits for using our struct. We have already
+added `TypeInfo` in order to give our struct access to this trait.
+
+For type `Gender`, we will need to build out our own custom enum and helper functions. Now is a good time 
+to do that.
+
+### Gender enum
+
+We have just created a struct that requires a custom type called `Gender`. This type will handle an enum 
+defining our Kitty's gender. To create it, you'll build out the following parts:
+
+* An enum declaration, which specifies Male and Female values.
+* Implement a helper function for our Kitty struct.
+
+1. Declare the custom enum
+
+Replace "TODO: Enum that represents the gender of a kitty" with the following code:
+
+```rust
+// Represents the gender of a kitty.
+	#[derive(Clone, Encode, Decode, PartialEq, Copy, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+	// We need this to pass kitty info for genesis configuration
+	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+	pub enum Gender {
+		Male,
+		Female,
```

Comment:
Add another geneder "Other". we are in 2023 :) and see if other code is affected